+MonteCarlo/@ParticleWalker/private/getLimitedStep.m

```diff
@@ -17,7 +17,8 @@
     end
 
     % update BEFORE checking new needUpdate (because of initialisation)
-    dxdydz = getStep(dim, varargin{:});
+    mask = contains({'x', 'y', 'z'}, cellstr(dim.'));
```

Comment:
Re-calculating the mask here inside the while loop isn't optimal, but I'm fine with leaving it for now. Maybe (eventually) move this outside the getLimitedStep altogether into `Substrate`:

```matlab
methods
  function obj = set.mask(obj, dim)
    if ischar(dim)
        obj.mask = contains({'x', 'y', 'z'}, cellstr(dim.'));
    else % assume correct input format
        obj.mask = dim;
  end
end
```